modules/assim.sequential/R/Remote_helpers.R

```diff
@@ -0,0 +1,440 @@
+  #
+  #Filter all the obs just for the sites we are simulating
+  point_list$median_AGB <-
+    point_list$median_AGB[[1]] %>% filter(Site_ID %in% site.ids)
+  point_list$stdv_AGB  <-
+    point_list$stdv_AGB[[1]] %>% filter(Site_ID %in% site.ids)
+  
+  #Finding the orders
+  site.order <-
+    sapply(site.ids, function(x)
+      which(point_list$median_AGB$Site_ID %in% x))  %>% unlist() %>% as.numeric() %>% na.omit()
+  #Reordering
+  point_list$median_AGB <- point_list$median_AGB[site.order, ]
+  point_list$stdv_AGB <- point_list$stdv_AGB[site.order, ]
+  
+  # truning lists to dfs  for both mean and cov
+  date.obs <-
+    strsplit(names(point_list$median_AGB), "_")[3:length(point_list$median_AGB)] %>%
+    map_chr( ~ .x[2]) %>% paste0(., "/12/31")
+  
+  #Making in a format that we need
+  obs.mean <-
+    names(point_list$median_AGB)[3:length(point_list$median_AGB)] %>%
+    map(function(namesl) {
+      ((point_list$median_AGB)[[namesl]] %>%
+         map( ~ .x %>% as.data.frame %>% `colnames<-`(c('AbvGrndWood'))) %>%
+         setNames(site.ids[1:length(.)])
+      )
+    }) %>% setNames(date.obs)
+  
+  
+  
+  obs.cov <-
+    names(point_list$stdv_AGB)[3:length(point_list$median_AGB)] %>%
+    map(function(namesl) {
+      ((point_list$stdv_AGB)[[namesl]] %>%
+         map(~ (.x) ^ 2 %>% as.matrix()) %>%
+         setNames(site.ids[1:length(.)]))
+      
+    }) %>% setNames(date.obs)
+  
+  
+  
+  return(list(obs.mean = obs.mean,
+              obs.cov = obs.cov))
+}
+
+
+
+#' SDA_remote_launcher
+#'
+#' @param settingPath The Path to the setting that will run SDA
+#' @param ObsPath  Path to the obs data which is expected to be an .Rdata.
+#'
+#' @export
+#' @return This function returns a list of two pieces of information. One the remote path that SDA is running and the PID of the active run.
+#' @example 
+#' \dontrun{
+#'  # This example can be found under inst folder in the package
+#'  library(PEcAn.all)
+#'  library(purrr)
+#'  
+#'  run.bash.args <- c(
+#'   "#$ -l h_rt=48:00:00",
+#'   "#$ -pe omp 28 # Request a parallel environment with 4 cores",
+#'   "#$ -l mem_per_core=1G # and 4G memory for each",
+#'   "#$ -l buyin",
+#'   "module load R/3.5.2",
+#'   "module load python/2.7.13"
+#'   )
+#'  settingPath <-"pecan.SDA.4sites.xml"
+#'
+#'  ObsPath <- "Obs/LandTrendr_AGB_output50s.RData"
+#'    
+#'  SDA_remote_launcher(settingPath, ObsPath, run.bash.args)
+#'}
+#'
+SDA_remote_launcher <-function(settingPath, 
+                               ObsPath,
+                               run.bash.args){
+
+  #---------------------------------------------------------------
+  # Reading the settings
+  #---------------------------------------------------------------
+  settings <- read.settings(settingPath)
+  my_host <- list(name =settings$host$name , tunnel = settings$host$tunnel, user=settings$host$user)
+  local_path <-settings$outdir
+  if (is.null(run.bash.args)) run.bash.args <-""
```

Comment:
Or better as set to NULL?

Reply:
I use this inside a  `writeLines` and NULL will result in 
```
Error in writeLines(NULL, paste0(tmpdir, "/Run.bash")) : 
  invalid 'text' argument
```